zebra-rpc/src/methods.rs

```diff
@@ -902,16 +901,19 @@ where
                 })?;
 
             match response {
-                zebra_state::ReadResponse::Transaction(Some((tx, height))) => Ok(
-                    GetRawTransaction::from_transaction(tx, Some(height), verbose != 0).map_err(
-                        |error| Error {
-                            code: ErrorCode::ServerError(0),
-                            message: error.to_string(),
-                            data: None,
-                        },
-                    )?,
-                ),
-                zebra_state::ReadResponse::Transaction(None) => Err(Error {
+                zebra_state::ReadResponse::Transaction {
+                    transaction_and_height: Some((tx, height)),
+                    confirmations: Some(confirmations),
```

Comment:
This might be easier to handle (and review) as an optional struct with 3 non-optional fields. Otherwise, there are combinations of optional fields that can never occur, but callers still have to handle them.

Reply:
Agreed, updated in [`6cb2823`](https://github.com/ZcashFoundation/zebra/pull/6287/commits/6cb2823c015f866aecf139902ce6b192ced99a10).